pkg/services/sqlstore/migrations/dashboard_panels_mig.go

```diff
@@ -0,0 +1,103 @@
+package migrations
+
+import (
+	"encoding/json"
+	"strconv"
+
+	"xorm.io/xorm"
+
+	"github.com/grafana/grafana/pkg/services/sqlstore/migrator"
+)
+
+func addDashboardPanelMaxDataPointsAsIntegerMigration(mg *migrator.Migrator) {
+	mg.AddMigration("update maxDataPoints value type as integer in dashboard panels", &MakePanelMaxDataPointsAsIntegerMigration{})
+}
+
+type MakePanelMaxDataPointsAsIntegerMigration struct {
+	migrator.MigrationBase
+}
+
+func (m *MakePanelMaxDataPointsAsIntegerMigration) SQL(dialect migrator.Dialect) string {
+	return "code migration"
+}
+
+type DashboardDataDTO struct {
+	Id   int64
+	Data string
+}
+
+func (m *MakePanelMaxDataPointsAsIntegerMigration) Exec(sess *xorm.Session, mg *migrator.Migrator) error {
+	dashboardDataDTOs := make([]*DashboardDataDTO, 0)
+
+	err := sess.SQL(`SELECT id, data FROM dashboard WHERE uid="f0532a5b-a3d9-43b6-8a4b-dab0ccf8066a"`).Find(&dashboardDataDTOs)
```

Comment:
This approach seems great... BUT having implemented the same thing for search, I know for a fact it will OOM and fail for ~2% of our instances.  

Given that we removed the error in https://github.com/grafana/grafana/pull/85562 -- I think we can postpone the dashboard cleanup till we have a better strategy for it.  (we will need one in general!  this is one of many patterns like this we want to be able to fix)

Reply:
Yeah migration is not ideal but still wanna put it in the PR to get feedback and maybe some suggestions to reach the same effect. Then I am gonna remove migration.